pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -434,6 +439,9 @@ func (plugin *kubenetNetworkPlugin) SetUpPod(namespace string, name string, id k
 			// Not a hard error or warning
 			glog.V(4).Infof("Failed to clean up %s/%s after SetUpPod failure: %v", namespace, name, err)
 		}
+
+		// Perform garbage collection on allocated ips
+		plugin.ipamGarbageCollection()
```

Comment:
Nope. This only happens when SetUpPod fails. Usually it is caused by either IP exhaust or infra container abnormally.